Source/Core/InputCommon/ControllerEmu/ProfileManager.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "InputCommon/DeviceProfileContainer.h"
+#include "InputCommon/InputProfile.h"
+
+#include <optional>
+#include <set>
+
+namespace InputCommon
+{
+class ProfileManager
+{
+public:
+  explicit ProfileManager(ControllerEmu::EmulatedController* owning_controller);
```

Comment:
You'd likely want to default the destructor in the cpp file as well, given the class contains a non-trivial type (std::set)